vendoring in tarball sources

onur-ozkan opened the pull request:
fixes #94782

onur-ozkan committed:
vendor distribution on the tarball sources

Signed-off-by: onur-ozkan <work@onurozkan.dev>

bjorn3 commented on src/bootstrap/dist.rs:
```diff
@@ -1001,11 +1001,16 @@ impl Step for PlainSourceTarball {
+            if builder.rust_info().is_managed_git_subrepository() {
+                // Ensure we have the submodules checked out.
+                builder.update_submodule(Path::new("src/tools/cargo"));
+                builder.update_submodule(Path::new("src/tools/rust-analyzer"));
```
rust-analyzer is no longer a submodule.

onur-ozkan committed:
remove the use of `fn update_submodule` on rust-analyzer

We don't need to run `fn update_submodule` on rust-analyzer
as it's no longer a submodule.

Signed-off-by: onur-ozkan <work@onurozkan.dev>

onur-ozkan commented:
a reminder ping @clubby789

bors commented:
:pushpin: Commit 92ab93fcb555d30cc88a994c98ba28244f0553f8 has been approved by `clubby789`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=clubby789 92ab93fcb555d30cc88a994c98ba28244f0553f8 -->
<!-- homu: {"type":"Approved","sha":"92ab93fcb555d30cc88a994c98ba28244f0553f8","approver":"clubby789","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->
